test/index.ts

```diff
@@ -832,6 +833,16 @@ function testNonEmptyObject() {
   type TestB = Assert<IsExact<NonEmptyObject<EmptyObject>, never>>;
 }
 
+function testNonEmptyArray() {
+  type TestA<T> = Assert<IsExact<NonEmptyArray<T>, [T, ...T[]]>>;
+  type TestB<T> = AssertFalse<IsExact<NonEmptyArray<T>, []>>;
+
+  type TestEmptyArrayNotAssignable<T> = AssertFalse<Assignable<NonEmptyArray<T>, []>>;
+  type TestArrayOneElementAssignable<T> = Assert<Assignable<NonEmptyArray<T>, [T]>>;
+  type TestArrayTwoElementsAssignable<T> = Assert<Assignable<NonEmptyArray<T>, [T, T]>>;
+  type TestArrayMoreThanOneElementsAssignable<T> = Assert<Assignable<NonEmptyArray<T>, [T, ...T[]]>>;
```

Comment:
I know that examples in this file are not well structured, but could you please change it to:

```ts
type cases = [
  Assert<IsExact<NonEmptyArray<T>, [T, ...T[]]>>,
  AssertFalse<IsExact<NonEmptyArray<T>, []>>,
  ....
];
```

I think it contains less types that we don't use anyway. And if you want to highlight it somehow, you can write comment right before the test case

WDYT?